### PR TITLE
Numpy 2.0 allowed in pyproject.toml

### PR DESCRIPTION
**Pull request by Ipuch**
### Related
* Part of #6762

### What
A very small commit to see if CI still pass with Numpy 2.0 and over allowed, as python projects are moving quickly to it.

### CODE REVIEW

File: rerun_py/pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ classifiers = [
 dependencies = [
   # Must match list in `.github/workflows/reusable_test_wheels.yml`
   "attrs>=23.1.0",
-  "numpy>=1.23,<2",
+  "numpy>=1.23",
```
Comment (abey79) [inline rerun_py/pyproject.toml:17]: This invalidates the `pixi.lock` file, which explains the CI failure. Could you run any pixi command (e.g. `pixi run rerun`) to update the lock file and push it?

Comment (Ipuch) [reply rerun_py/pyproject.toml:17]: Done.

**Comment by teh-cmc** [comment]
@rerun-bot approve

**Comment by abey79** [comment]
It looks like you are using a pixi version that is newer than the one we currently have in CI. I'll check if we can migrate to the new version to make this PR mergeable.

**Comment by abey79** [comment]
- https://github.com/rerun-io/rerun/pull/8320

**Comment by Ipuch** [comment]
Can I update my branch?

**Comment by abey79** [comment]
> Can I update my branch?

Yes, should be good now.

**Comment by Ipuch** [comment]
C++ CI failed

**Pull request merged**